app/controllers/mixins/actions/vm_actions/reconfigure.rb

```diff
@@ -216,13 +216,13 @@ def get_vlan_options(host_id)
           # determine available switches for this host...
           switch_ids = []
           Rbac.filtered(HostSwitch.where("host_id = ?", host_id)).each do |host_switch|
-            switch_ids << host_switch.switch_id
+            switch_ids << host_switch.switch_id unless switch_ids.include?(host_switch.switch_id)
```

Comment:
@jeffibm is this method even being called for ovirt vm reconfigure?

Reply:
its coming in from [`@vlan_options = vm.try(:available_vlans) || get_vlan_options(vm.host_id)`](https://github.com/ManageIQ/manageiq-ui-classic/blob/cc269e54618e6bf43f1ee4b54bc9314225db7879/app/controllers/mixins/actions/vm_actions/reconfigure.rb#L22)

Reply:
none of my records were hitting the `get_vlan_options` method. that method looked like it could have duplicates :)

Also, when I checked with the `vm.available_vlans`, it's executing a set of queries and returns an array. Not sure how to refine this relation.

Or, should we just add` .uniq` to this line in [reconfigure.rb](https://github.com/ManageIQ/manageiq-ui-classic/blob/cc269e54618e6bf43f1ee4b54bc9314225db7879/app/controllers/mixins/actions/vm_actions/reconfigure.rb#L22), like
`@vlan_options = (vm.try(:available_vlans) || get_vlan_options(vm.host_id)).uniq`